OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -101,7 +103,9 @@ void INotifyCreated.Created(Actor self)
 
 		void INotifyDamage.Damaged(Actor self, AttackInfo e)
 		{
-			if (Info.UncloakOn.HasFlag(UncloakType.Damage))
+			var isHeal = e.Damage.Value < 0;
```

Comment:
Honestly that check should be put in when INotifyDamaged is sent out, it makes no sense that the notification is sent out when there is no change to the health of the unit.